### ISSUE TITLE
Create a basic guiding doc - How to Debug Your Airflow Deployment

### ISSUE THREAD

**Issue by omkar-foss** [apache/airflow#43892]
### Description

Create a basic guiding doc "How to Debug Your Airflow Deployment" that can guide users towards debugging their Airflow deployment. This doc can be based on general Airflow best practices + Agans' 9 rules of debugging:

1. Understand the system
2. Make it fail
3. Quit thinking and look
4. Divide and conquer
5. Change one thing at a time
6. Keep an audit trail
7. Check the plug
8. Get a fresh view
9. If you didn't fix it, it ain't fixed

The guide shall express above rules using Airflow examples, such that the users can read the guide and understand how to approach similar Airflow debugging scenarios.

### Use case/motivation

David A. Wheeler's Article (2004): https://dwheeler.com/essays/debugging-agans.html

### Related issues

Parent Issue: https://github.com/apache/airflow/issues/40975

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Dev-iL** [comment]
[Relevant discussion on Slack](https://apache-airflow.slack.com/archives/C07J87PK1BK/p1731346161804209) with some ideas on how the outputs of this should be structured and how to set the readers' expectations.

**Comment by KoviAnusha** [comment]
@omkar-foss are you still working on this? If not, can I work on this?

cc: @potiuk

**Comment by omkar-foss** [comment]
> [@omkar-foss](https://github.com/omkar-foss) are you still working on this? If not, can I work on this?
> 
> cc: [@potiuk](https://github.com/potiuk)

Sure @KoviAnusha, please feel free to pick this up, I haven't found the time to work on this. Assigning this issue to you, thanks.

**Comment by KoviAnusha** [comment]
Thank you

**Comment by rawwar** [comment]
I'm not sure if we need this. We have https://github.com/apache/airflow/blob/main/contributing-docs/20_debugging_airflow_components.rst

Maybe we can extend on it.



EDIT: 

I have been trying to debug every component to see if its possible to debug any part of the code. 

So far, I was able to completely debug dag-processor line-by-line, including the part where we hand it off to to task-sdk.  I was also able to walk through most of the api-server and celery worker(I still haven't looked into the part where the celery fork pool worker runs the task. But, that's something i will be doing soon).  

I plan to write  a doc on my perspective of debugging airflow components. But, that's going to take sometime as I want to do that myself and see if I can improve it and then document it.

**Comment by omkar-foss** [comment]
>I'm not sure if we need this. We have https://github.com/apache/airflow/blob/main/contributing-docs/20_debugging_airflow_components.rst

Yes but currently the doc is breeze-specific, I suppose we could extend it further.

>I plan to write a doc on my perspective of debugging airflow components. But, that's going to take sometime as I want to do that myself and see if I can improve it and then document it.

@rawwar did you get a chance to continue it? If not let me know, I'll have a look.
